kubernetes/resource_kubernetes_job_v1.go

```diff
@@ -48,6 +51,51 @@ func resourceKubernetesJobV1() *schema.Resource {
 	}
 }
 
+func resourceKubernetesJobV1CustomizeDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+	if d.Id() == "" {
+		log.Printf("[DEBUG] Resource ID is empty, resource not created yet.")
+		return nil
+	}
+
+	// Retrieve old and new TTL values
+	oldTTLRaw, newTTLRaw := d.GetChange("spec.0.ttl_seconds_after_finished")
+
+	// If both TTL values are not set or are empty, skip TTL diff
+	if (oldTTLRaw == nil && newTTLRaw == nil) || (oldTTLRaw == "" && newTTLRaw == "") {
+		log.Printf("[DEBUG] No ttl_seconds_after_finished provided or both values are empty; skipping TTL diff")
+		return nil
+	}
+
+	// Only check TTL if present
+	var oldTTLStr, newTTLStr string
+	if oldTTLRaw != nil {
+		oldTTLStr = oldTTLRaw.(string)
+	}
+	if newTTLRaw != nil {
+		newTTLStr = newTTLRaw.(string)
+	}
+
+	oldTTLInt, err := strconv.Atoi(oldTTLStr)
+	if err != nil {
+		log.Printf("[DEBUG] Invalid old TTL value: %s, skipping diff", oldTTLStr)
+		return nil
+	}
+	newTTLInt, err := strconv.Atoi(newTTLStr)
+	if err != nil {
+		log.Printf("[DEBUG] Invalid new TTL value: %s, skipping diff", newTTLStr)
+		return nil
+	}
+
+	// Suppress the diff if the old and new TTL values are the same
+	if oldTTLInt == newTTLInt {
+		log.Printf("[DEBUG] ttl_seconds_after_finished has not changed; suppressing diff")
+		d.Clear("spec")
+		d.Clear("metadata")
+	}
+
+	return nil
+}
```

Comment:
It doesn't make sense to me why we would have a diff function that checks if the value of `ttl_seconds_after_finished` hasn't changed and then suppress the diff? 

I tried removing this function completely and your tests still pass.